channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -348,21 +360,30 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControls {
                     let customInterface = senderConfig.httpXHROverride;
                     let httpInterface: IXHROverride = null;
                     let syncInterface: IXHROverride = null;
-                    httpInterface = _getSenderInterface([TransportType.Xhr, TransportType.Fetch], false);
+
+                    // Prefix any user requested transport(s) values
+                    let theTransports: TransportType[] = _prependTransports([TransportType.Xhr, TransportType.Fetch], senderConfig.transports);
+
+                    if (senderConfig.isBeaconApiDisabled || !isBeaconsSupported()){
```

Comment:
Lets not do all of this here as it's a lot of code for no real  gain.

Please add an additional if check `senderConfig.isBeaconApiDisabled` to the `_getSenderInterface` for it to select the `beacon` transport. 
This will also help with the bundle size (one of the reasons the tests are failing), the above will be a lot smaller.